.github/workflows/commit-main.yml

```diff
@@ -21,69 +23,86 @@ jobs:
     - name: Get dependencies
       run: make deps
 
-    - name: Build
-      run: make build-local
-
-  test:
-    name: Test
-    runs-on: ubuntu-latest
-    steps:
+    - name: Compile
+      run: make compile
 
-    - name: Set up Go 1.x
-      uses: actions/setup-go@v2
+    - name: Archive production artifacts
+      uses: actions/upload-artifact@v2
       with:
-        go-version: ^1.13
-
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
-
-    - name: Get dependencies
-      run: make deps
+        name: worklog-binary
+        path: bin
 
-    - name: Create folders
-      run: mkdir ~/.worklog && mkdir ~/.worklog-backup && touch ~/.worklog/a
+  run-ubuntu:
+    name: Run-ubuntu
+    runs-on: ubuntu-latest
+    needs: [store]
 
-    - name: Test once
-      run: make test
+    strategy:
+      matrix:
+        binary:
+          - "32bit/worklog-linux"
+          - "64bit/worklog-linux"
 
-  format:
-    name: Format
-    runs-on: ubuntu-latest
     steps:
-
-    - name: Set up Go 1.x
-      uses: actions/setup-go@v2
+    - name: Download
+      uses: actions/download-artifact@v2
       with:
-        go-version: ^1.13
+        name: worklog-binary
 
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
+    - name: Permissions
+      run: "sudo chown -R $USER:$USER ./32bit ./64bit && sudo chmod +x 32bit/worklog-* 64bit/worklog-*"
 
-    - name: Check formatting
-      run: make format
+    - name: Extract
+      run: "mv ./${{ matrix.binary }} ./worklog"
 
-  store:
-    name: Store artifacts
-    runs-on: ubuntu-latest
-    needs: [build, test, format]
+    # TODO change to tests
+    - name: Test
+      run: "./worklog"
+
+  run-mac:
+    name: Run-mac
+    runs-on: macos-latest
+    needs: [store]
+
+    strategy:
+      matrix:
+        binary:
+          - "64bit/worklog-darwin"
 
     steps:
-    - name: Set up Go 1.x
-      uses: actions/setup-go@v2
+    - name: Download
+      uses: actions/download-artifact@v2
       with:
-        go-version: ^1.13
+        name: worklog-binary
 
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
+    - name: Permissions
+      run: "sudo chown -R $USER ./32bit ./64bit && sudo chmod +x 32bit/worklog-* 64bit/worklog-*"
 
-    - name: Get dependencies
-      run: make deps
+    - name: Extract
+      run: "mv ./${{ matrix.binary }} ./worklog"
 
-    - name: Compile
-      run: make compile
+    - name: Test
+      run: "./worklog"
 
-    - name: Archive production artifacts
-      uses: actions/upload-artifact@v2
+  run-windows:
+    name: Run-windows
+    runs-on: windows-latest
+    needs: [store]
+
+    strategy:
+      matrix:
+        binary:
+          - "32bit\\worklog-windows"
+          - "64bit\\worklog-windows"
```

Comment:
These should have `.exe` for windows.
Will probably need changing in the Makefile as well.